src/Morphir/Elm/Backend/Codec/Gen.elm

```diff
@@ -0,0 +1,278 @@
+module Morphir.Elm.Backend.Codec.Gen exposing (..)
+
+import Elm.Syntax.Declaration exposing (Declaration(..))
+import Elm.Syntax.Expression exposing (Case, Expression(..), Function, FunctionImplementation)
+import Elm.Syntax.ModuleName exposing (ModuleName)
+import Elm.Syntax.Node exposing (Node(..))
+import Elm.Syntax.Pattern exposing (Pattern(..), QualifiedNameRef)
+import Elm.Syntax.Range exposing (emptyRange)
+import Morphir.IR.AccessControlled exposing (AccessControlled(..))
+import Morphir.IR.Advanced.Type exposing (Constructor, Definition(..), Field(..), Type(..), field, record)
+import Morphir.IR.FQName exposing (FQName(..))
+import Morphir.IR.Name exposing (Name, fromString, toCamelCase, toTitleCase)
+import Morphir.IR.Path as Path exposing (toString)
+
+
+typeDefToEncoder : extra -> Name -> AccessControlled (Definition extra) -> Declaration
+typeDefToEncoder e typeName typeDef =
+    let
+        function : Function
+        function =
+            { documentation = Nothing
+            , signature = Nothing
+            , declaration = emptyRangeNode functionImpl
+            }
+
+        functionImpl : FunctionImplementation
+        functionImpl =
+            { name = emptyRangeNode functionName
+            , arguments = args
+            , expression = emptyRangeNode funcExpr
+            }
+
+        functionName : String
+        functionName =
+            [ "encode" ] ++ typeName |> toCamelCase
```

Comment:
Can you do `Name.toCamelCase` as discussed?

Reply:
Oh my bad!